packages/syft/src/syft/types/newdatetime.py

```diff
@@ -0,0 +1,45 @@
+# stdlib
+from datetime import datetime, timezone
+from functools import total_ordering
+from typing import Any
+
+# third party
+from typing_extensions import Self
+import pydantic
+
+# relative
+from ..serde.serializable import serializable
+from datetime import datetime
+
+@serializable()
+@total_ordering
+class NewDateTime(pydantic.BaseModel):
```

Comment:
If the old Datetime was a syft object with a version then you can rename the class since the lookup is from the canonical name, which means you can call this one `DateTime`.

I do still think removing the lookup / versioning aspect of this type just means that we corner ourselves for future changes.

This could easily be fixed by adding another object to the .mro()

SyftObjectBase(pydantic.BaseModel, SyftHashableObject) <- no versioning
SyftObjectVersioned(SyftObjectBase) <- new class with __canonical_name__ and __version__
SyftObject(SyftObjectVersioned) <- all the crazy new fields that are being added mostly for our larger Document store style objects, which I agree feels overkill for a DateTime object.

```
# the old class
class DateTimeV2(SyftObject):
    __canonical_name__ = "DateTime"
    __version__ = SYFT_OBJECT_VERSION_2

# your new class here
class DateTime(SyftObjectVersioned):
    __canonical_name__ = "SyftObject"
    __version__ = SYFT_OBJECT_VERSION_3
```

@koenvanderveen thoughts?

Reply:
This new interface would indeed fix this problem, but at the same time we are currently looking into how to properly handle third-party objects in serialization as in the migration PR we have started to use canonical names instead of fully qualified names. This could be a solution for versioning as well, by adding an exception for `NewDateTime` without it having to inherit from `SyftObject`